src/ngff_tables_prototype/reader.py

```diff
@@ -9,52 +12,396 @@
 from napari.types import LayerDataTuple
 import pandas as pd
 import zarr
+from typing import Union, Optional
+import warnings
+import dask.array as da
 
 
 def load_table_to_anndata(file_path: str, table_group: str) -> AnnData:
     return read_zarr(os.path.join(file_path, table_group))
 
 
-def _anndata_to_napari_points(anndata_obj: AnnData) -> LayerDataTuple:
-    points_coords = anndata_obj.X
-    layer_properties = anndata_obj.obs
-    return points_coords, {'properties': layer_properties}, 'points'
-
-def _anndata_to_napari_labels_features(anndata_obj: AnnData) -> pd.DataFrame:
+def _anndata_to_napari_labels_features(
+    anndata_obj: AnnData, table_instance_key: str
+) -> pd.DataFrame:
     df = pd.concat([anndata_obj.to_df(), anndata_obj.obs], axis=1)
 
-    # hacky add dummy data to first row to meet the condition for napari features
+    # hacky add dummy data to first row to meet the condition for napari features (first row is the background)
     # https://github.com/napari/napari/blob/2648c307168fdfea7b5c03aaad4590972e5dfe2c/napari/layers/labels/labels.py#L72-L74
-    first_row = df.iloc[0, :]
-    first_row["cell_id"] = 0
+    warnings.warn(
+        "background value is arbitrary, this is just for testing purposes and will have to be changed"
+    )
+    first_row = df.iloc[0, :].copy()
+    first_row[table_instance_key] = 0
+    return pd.concat([pd.DataFrame(first_row).T, df])
 
-    return pd.concat([first_row, df])
 
-def load_to_napari_layer_data(file_path: str, table_group_key: str):
+def get_napari_image_layer_data(file_path: str) -> LayerDataTuple:
     ome_zarr = parse_url(file_path)
     reader = Reader(ome_zarr)
     reader_func = transform(reader())
     layer_data = reader_func(file_path)
+    # here we consider only images, as we deal with labels and other types in dedicated functions
+    image_layer_data = []
+    for layer in layer_data:
+        data, kwargs, layer_type = layer
+        if layer_type == "image":
+            # this doesn't work, so the RGB image is split into three channels
+            # first element of the pyramidal array
+            # if data[0].shape[0] in [3, 4]:
+            #     kwargs['rgb'] = True
+            # assuming y and x are the two last dimesions
+            image_layer_data.append(
+                ([da.moveaxis(d, -1, -2) for d in data], kwargs, layer_type)
+            )
+    return image_layer_data
+
+
+def _colorize_regions_for_napari_layer(
+    features_table,
+    kwargs,
+    color_key_name: str,
+    tables_instance_key: Optional[str] = None,
+):
+    """helper function to build a list of colors to colorize some regions
```

Comment:
we have some tooling in napari to do this, but I think we can address that later. this looks good for now!